src/components/display/formFooter/testConnectionResult.tsx

```diff
@@ -57,11 +60,12 @@ export function FormFooterTestConnectionResult({
   }, [isLoading]);
 
   useEffect(() => {
-    if (!isLoading) {
+    if (!isLoading && !isSubmitting) {
       progressElemRef.current
         ?.querySelector(`.${linearProgressClasses.bar}`)
         ?.addEventListener('transitionend', () => {
-          window.setTimeout(() => {
+          clearTimeout(timeoutId.current);
+          timeoutId.current = window.setTimeout(() => {
             if (failureError) {
               setErrorMessage(failureError.message || 'Unknown error');
             } else {
```

Comment:
Since you are here, may I ask to replace content to this:
```tsx
let hasTransitionEnded = false;

const handleTransitionEnd = () => {
  setTimeout(() => {
    if (!hasTransitionEnded) {
      hasTransitionEnded = true;
      if (failureReason) {
        setError(
          failureReason || 'Connection failed, please check your configuration and try again',
        );
      } else {
        setSuccess(true);
        onSuccess();
      }
    }
  }, 300);
};

const progressBar = progressElemRef.current?.querySelector(`.${linearProgressClasses.bar}`);

progressBar?.addEventListener('transitionend', handleTransitionEnd, { once: true });
progressBar?.addEventListener('webkitTransitionEnd', handleTransitionEnd, { once: true });

clearTimeout(intervalId.current);
setProgress(100);

return () => {
  progressBar?.removeEventListener('transitionend', handleTransitionEnd);
  progressBar?.removeEventListener('webkitTransitionEnd', handleTransitionEnd);
};
```

I just noticed in some cases the transitionend is not fired in the Safari on e2e tests